core-backend/sandbox/src/env.rs

```diff
@@ -46,16 +46,18 @@ use sp_sandbox::{
 
 #[derive(Debug, derive_more::Display)]
 pub enum SandboxEnvironmentError {
-    #[display(fmt = "Unable to instantiate module")]
-    ModuleInstantiation,
-    #[display(fmt = "Unable to get wasm module exports")]
-    GetWasmExports,
+    #[display(fmt = "Unable to instantiate module: {:?}", _0)]
+    ModuleInstantiation(sp_sandbox::Error),
+    #[display(fmt = "Unable to get wasm module exports: {}", _0)]
+    GetWasmExports(String),
     #[display(fmt = "Unable to set module memory data")]
     SetModuleMemoryData,
     #[display(fmt = "Unable to save static pages initial data")]
     SaveStaticPagesInitialData,
-    #[display(fmt = "Failed to create env memory")]
-    CreateEnvMemory,
+    #[display(fmt = "Failed to create env memory: {:?}", _0)]
+    CreateEnvMemory(sp_sandbox::Error),
+    #[display(fmt = "No trap explanation")]
+    NoTrapExplanation,
```

Comment:
There should be no cases without trap explanation. Our backend field keeps zero only for cases of `unreachable` section in wasm. This should be checked somehow and research if there are similar-behaviour reasons